google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -145,16 +149,33 @@ public CancellableRunnable createDeployRunner(
         loggingHandler,
         deploymentConfiguration,
         targetEnvironment,
-        wrapCallbackForUsageTracking(callback, deploymentConfiguration, targetEnvironment));
+        wrapCallbackForUsageTracking(callback, targetEnvironment));
 
-    boolean isFlexCompat = targetEnvironment.isFlexible()
-        && AppEngineProjectService.getInstance().isFlexCompat(project, source);
-    if (targetEnvironment.isStandard() || isFlexCompat) {
-      return createStandardRunner(loggingHandler, Paths.get(source.getFilePath()), deploy,
-          isFlexCompat);
+    if (targetEnvironment.isStandard() || targetEnvironment.isFlexCompat()) {
```

Comment:
Not too familiar with that code branch, but it seems that we're creating a fictional module "userSpecifiedSource".
I agree, though, that this isn't new and should be addressed in another PR.